torch/stable/library.h

```diff
@@ -0,0 +1,149 @@
+// this file can only have stable stuff! Akin to shim.h
+// but unlike shim.h, this file can contain header-only C++
+// code for better UX.
+
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+
+
+template <typename T>
+StableIValue from(T val) {
+  static_assert(
+      sizeof(T) <= sizeof(StableIValue),
+      "StableLibrary stack does not support parameter types larger than 64 bits.");
+  return *reinterpret_cast<StableIValue*>(&val);
+}
+
+template <typename T>
+T to(StableIValue val) {
+  return *reinterpret_cast<T*>(&val);
+}
```

Comment:
It looks cool, but templates like that are by design unstable, as they will produce different symbols depending on whether or not you compile your code with `-frtti` vs `-fno-rtti` , `-fexceptions`/`-fnoexceptions` and perhaps so on. If you really want a stable API(and ABI i.e. compile once use across versions of torch) you'll need C api, or something that explicitly decorated as `nothrow` (though this has to be checked)